specification/sbi_atee.adoc

```diff
@@ -938,6 +938,202 @@ The possible error codes returned in `sbiret.error` are shown below.
+* The page is in either the `Blocked` state and uniquely owned by the TVM, or in the `BlockedShared`
+state and owned by the host.
+* The TLB version at which the page was invalidated is older than the TVM’s current TLB version.
+There is a free page in the TVM’s page-table page pool.
+
+After verifying these pre-conditions are met, the TSM:
+
+* Allocates and fills in a page-table page mapping the sub-pages of the huge page to be split,
+marking each PTE as present.
+* Installs a non-leaf PTE pointing the newly-allocated page-table page in place of the huge-page
+leaf PTE.
+* Places all the sub-pages in the `Mapped` state (if `Blocked`) or `Shared` state (if `BlockedShared`).
+If the page was `BlockedShared`, the reference count of the page is restored.
+
+The possible error codes returned in `sbiret.error` are shown below.
+
+[#table_sbi_tee_host_tvm_demote_page_errors]
+.TEE Host TVM Demote Page
+[cols="2,3", width=90%, align="center", options="header"]
+|===
+| Error code              | Description
+| SBI_SUCCESS             | The operation completed successfully.
+| SBI_ERR_INVALID_PARAM   | `tvm_guest_id` or `base_page_address` or `tsm_page_type` were invalid.
+| SBI_ERR_FAILED          | The operation failed for unknown reasons.
+|===
+
+[#sbi_tee_host_tvm_remove_pages]
+=== Function: TEE Host TVM Remove Pages (FID #20)
+[source, C]
+-----
+struct sbiret sbi_tee_host_tvm_remove_pages(unsigned long tvm_guest_id,
+                                           unsigned long base_page_address,
+                                           unsigned long length);
+-----
+
+Removes mappings from a TVM. The range to be unmapped must already have been invalidated and fenced,
+and must lie within a removable region of the guest's physical address space.
+
+The TSM verifies that:
+
+* All pages within the requested range are invalidated and are in the `Blocked` or `BlockedShared` state.
+* The TLB version at which the pages were invalidated is older than the TVM’s current TLB version.
+* The address range lies within a removable (`SharedRemovable` or `ConfidentialRemovable`) region of guest
+physical address space.
+
+After verifying these pre-conditions are met, the TSM:
+
+* Clears (zeros out) all PTEs within the specified range.
+* Places each page in either the `Converted` state, with ownership returned to the host (if previously
+`Blocked`) or the `Shared` state, with the reference count decremented (if previously `BlockedShared`). 
+
+The possible error codes returned in `sbiret.error` are shown below.
+
+[#table_sbi_tee_host_tvm_remove_pages_errors]
+.TEE Host TVM Remove Pages
+[cols="2,3", width=90%, align="center", options="header"]
+|===
+| Error code              | Description
+| SBI_SUCCESS             | The operation completed successfully.
+| SBI_ERR_INVALID_PARAM   | `tvm_guest_id` or `base_page_address` or `length` were invalid.
+| SBI_ERR_FAILED          | The operation failed for unknown reasons.
+|===
```

Comment:
Invalid address error code if the address range doesn't lie within correct range.

Reply:
Same comment applies for all the functions.

Reply:
Commits removed. I will be opening a new PR for that after reworking the share/unshare parts.